File: go/01.go
package main

// Every solution is run along with utils.go
// e.g `go run 01.go utils.go`

import (
	"fmt"
	"strconv"
	"sort"
)

func getElfs(input_strings []string) []int {
	var elfs []int
	var elf int = 0
	for _, s := range input_strings {
		num, err := strconv.Atoi(s)
		if err == nil {
			elf += num
		} else {
			elfs = append(elfs, elf)
			elf = 0
		}
	}
	return elfs
}

func part_one(input_strings []string) int {
	return arrayMax(getElfs(input_strings))
}

func part_two(input_strings []string) int {
	var elfs []int = getElfs(input_strings)
	sort.Ints(elfs)
	return arraySum(elfs[len(elfs)-3:])
}

func main() {
	lines := getInputLines("data/01.txt")

	fmt.Println(part_one(lines))
	fmt.Println(part_two(lines))
}

File: python/01.py
from utils import get_input


def get_elfs(data):
    elfs = []
    elf = 0
    for calories in data:
        if calories != '':
            elf += int(calories)
        else:
            elfs.append(elf)
            elf = 0
    return elfs

def part_one(data):
    return max(get_elfs(data))

def part_two(data):
    return sum(sorted(get_elfs(data), reverse=True)[:3])


if __name__ == "__main__":
    path = 'data/01.txt'
    data = get_input(path)

    # First-part
    print(part_one(data))

    # Second-part
    print(part_two(data))


    # Shorter alternative
    elfs = [sum(map(int,elf.strip(',').split(','))) for elf in open('data/01.txt').read().replace('\n',',').split(',,')]
    part1 = max(elfs)
    part2 = sum(sorted(elfs, reverse=True)[:3])
    print(part1, part2)
